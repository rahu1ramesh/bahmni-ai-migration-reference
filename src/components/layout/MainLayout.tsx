import React, { ReactNode } from 'react';
import {
  Header,
  HeaderName,
  HeaderNavigation,
  HeaderMenuItem,
  SkipToContent,
} from '@carbon/react';

interface MainLayoutProps {
  children: ReactNode;
}

const MainLayout: React.FC<MainLayoutProps> = ({ children }) => {
  return (
    <>
      <Header aria-label="Bahmni Clinical">
        <SkipToContent />
        <HeaderName href="/" prefix="">
          Bahmni Clinical
        </HeaderName>
        <HeaderNavigation aria-label="Main Navigation">
          <HeaderMenuItem href="/">Home</HeaderMenuItem>
          {/* Add more navigation items as needed */}
        </HeaderNavigation>
      </Header>
      <main id="main-content">{children}</main>
    </>
  );
};

export default MainLayout;
