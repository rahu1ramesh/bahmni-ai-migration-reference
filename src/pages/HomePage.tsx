import React from 'react';
import { Grid, Column, Section } from '@carbon/react';
import PatientDetails from '@components/patient/PatientDetails';
import ConditionsTable from '@components/conditions/ConditionsTable';
import AllergiesTable from '@components/allergies/AllergiesTable';
import TreatmentsTable from '@components/treatments/TreatmentsTable';

const HomePage: React.FC = () => {
  return (
    <Section>
      <Grid>
        <Column lg={16} md={8} sm={4}>
          <PatientDetails />
          <AllergiesTable />
          <ConditionsTable />
          <TreatmentsTable />
        </Column>
      </Grid>
    </Section>
  );
};

export default HomePage;
