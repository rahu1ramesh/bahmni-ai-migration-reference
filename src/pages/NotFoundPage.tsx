import React from 'react';
import { Grid, Column, Heading, Section, Link } from '@carbon/react';
import { Link as RouterLink } from 'react-router-dom';

const NotFoundPage: React.FC = () => {
  return (
    <Section>
      <Grid>
        <Column lg={16} md={8} sm={4}>
          <Heading>404 - Page Not Found</Heading>
          <p>The page you are looking for does not exist.</p>
          <Link as={RouterLink} to="/">
            Return to Home
          </Link>
        </Column>
      </Grid>
    </Section>
  );
};

export default NotFoundPage;
