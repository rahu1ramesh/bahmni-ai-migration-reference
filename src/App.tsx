import React from 'react';
import { Routes, Route } from 'react-router-dom';
import { Content } from '@carbon/react';
import MainLayout from './components/layout/MainLayout';
import HomePage from './pages/HomePage';
import NotFoundPage from './pages/NotFoundPage';

const App: React.FC = () => {
  return (
    <MainLayout>
      <Content>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/clinical/:patientUuid" element={<HomePage />} />
          <Route path="*" element={<NotFoundPage />} />
        </Routes>
      </Content>
    </MainLayout>
  );
};

export default App;
