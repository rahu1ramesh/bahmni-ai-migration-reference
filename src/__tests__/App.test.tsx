import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';
import MainLayout from '@components/layout/MainLayout';
import HomePage from '../pages/HomePage';
import NotFoundPage from '../pages/NotFoundPage';

// Mock dependencies
jest.mock('@components/layout/MainLayout', () => {
  return jest.fn(({ children }) => (
    <div data-testid="mock-main-layout">{children}</div>
  ));
});

jest.mock('../pages/HomePage', () => {
  return jest.fn(() => <div data-testid="mock-home-page">Home Page</div>);
});

jest.mock('../pages/NotFoundPage', () => {
  return jest.fn(() => (
    <div data-testid="mock-not-found-page">Not Found Page</div>
  ));
});

jest.mock('@carbon/react', () => ({
  Content: jest.fn(({ children }) => (
    <div data-testid="mock-carbon-content">{children}</div>
  )),
}));

describe('App Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render without crashing', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    expect(screen.getByTestId('mock-main-layout')).toBeInTheDocument();
    expect(screen.getByTestId('mock-carbon-content')).toBeInTheDocument();
  });

  it('should render HomePage for root path', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>,
    );

    expect(HomePage).toHaveBeenCalled();
    expect(screen.getByTestId('mock-home-page')).toBeInTheDocument();
    expect(screen.queryByTestId('mock-not-found-page')).not.toBeInTheDocument();
  });

  it('should render HomePage for patient-specific path', () => {
    render(
      <MemoryRouter
        initialEntries={['/clinical/123e4567-e89b-12d3-a456-426614174000']}
      >
        <App />
      </MemoryRouter>,
    );

    expect(HomePage).toHaveBeenCalled();
    expect(screen.getByTestId('mock-home-page')).toBeInTheDocument();
    expect(screen.queryByTestId('mock-not-found-page')).not.toBeInTheDocument();
  });

  it('should render NotFoundPage for unknown paths', () => {
    render(
      <MemoryRouter initialEntries={['/unknown-path']}>
        <App />
      </MemoryRouter>,
    );

    expect(NotFoundPage).toHaveBeenCalled();
    expect(screen.getByTestId('mock-not-found-page')).toBeInTheDocument();
    expect(screen.queryByTestId('mock-home-page')).not.toBeInTheDocument();
  });

  it('should wrap content with MainLayout', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    expect(MainLayout).toHaveBeenCalled();
    const mainLayout = screen.getByTestId('mock-main-layout');
    expect(mainLayout).toBeInTheDocument();
    expect(mainLayout).toContainElement(
      screen.getByTestId('mock-carbon-content'),
    );
  });

  it('should render Routes component correctly', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    // Verify that the Routes component is rendered
    const contentElement = screen.getByTestId('mock-carbon-content');
    expect(contentElement).toBeInTheDocument();

    // Since Routes is not easily testable directly, we verify its behavior
    // by testing the rendered components based on different routes
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
