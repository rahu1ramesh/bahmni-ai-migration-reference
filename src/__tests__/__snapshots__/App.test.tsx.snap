// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="mock-main-layout"
  >
    <div
      data-testid="mock-carbon-content"
    >
      <div
        data-testid="mock-home-page"
      >
        Home Page
      </div>
    </div>
  </div>
</DocumentFragment>
`;
